src/main/java/org/broadinstitute/hellbender/tools/copynumber/datacollection/PerBaseCountCollector.java

```diff
@@ -0,0 +1,150 @@
+package org.broadinstitute.hellbender.tools.copynumber.datacollection;
+
+import htsjdk.samtools.util.Locatable;
+import org.broadinstitute.hellbender.tools.copynumber.formats.collections.PerBaseCountCollection;
+import org.broadinstitute.hellbender.tools.copynumber.formats.metadata.SampleLocatableMetadata;
+import org.broadinstitute.hellbender.tools.copynumber.formats.records.PerBaseCount;
+import org.broadinstitute.hellbender.utils.Nucleotide;
+import org.broadinstitute.hellbender.utils.SimpleInterval;
+import org.broadinstitute.hellbender.utils.Utils;
+import org.broadinstitute.hellbender.utils.param.ParamUtils;
+import org.broadinstitute.hellbender.utils.pileup.PileupElement;
+import org.broadinstitute.hellbender.utils.pileup.ReadPileup;
+
+import java.util.*;
+
+/**
+ * Collects per-base counts at specified sites where bases are one of those specified in {@link PerBaseCountCollector#BASES}.
+ * If two reads share the same query name and, therefore, originate from the same template of DNA, this
+ * only counts the higher base quality nucleotide of the two reads. If the base qualities of the two reads are equal,
+ * the nucleotide from read one will be counted.
+ *
+ * @author Samuel Lee &lt;slee@broadinstitute.org&gt;
+ * @author Robert Klein &lt;rklein@broadinstitute.org&gt;
+ */
+public class PerBaseCountCollector {
```

Comment:
Make class final.  Probably best to extract methods common to both this and `AllelicCountCollector` to an abstract class and just override the `collectAtLocus` method.